src/conn/conn_sweep.c

```diff
@@ -359,8 +366,14 @@ __wt_sweep_create(WT_SESSION_IMPL *session)
 	/*
 	 * Handle sweep does enough I/O it may be called upon to perform slow
 	 * operations for the block manager.
+	 *
+	 * The sweep thread sweeps the lookaside table for outdated records,
+	 * it gets its own cursor for that purpose.
+	 *
+	 * Don't tap the sweep thread for eviction.
 	 */
-	F_SET(session, WT_SESSION_CAN_WAIT);
+	F_SET(session, WT_SESSION_CAN_WAIT |
+	    WT_SESSION_LOOKASIDE_CURSOR | WT_SESSION_NO_EVICTION);
```

Comment:
There's a file named `evict_lru.c`, and it implements LRU eviction, I think it would be confusing to refer to the opportunistic case as LRU eviction. The point isn't if the thread can evict or not, the point is if the thread can be re-purposed from whatever it's doing to working on eviction at a particular point in time, right?

I'm happy to flip this around if that's a better approach.